src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3185,6 +3185,12 @@ private bool IsPossibleAccessorModifier()
             // to the user that this is not allowed.
             if (IsPossibleModifier())
             {
+                if (this.CurrentToken.Kind == SyntaxKind.RefKeyword)
```

Comment:
>if (this.CurrentToken.Kind == SyntaxKind.RefKeyword) [](start = 16, length = 52)

Perhaps rather than checking the token we should call ```private static DeclarationModifiers GetModifier(SyntaxToken token)``` instead of ```IsPossibleModifier``` and compare its return against ```DeclarationModifiers.Ref```?